envsource_assign_test.go

```diff
@@ -0,0 +1,359 @@
+		},
+		{
+			"WithMapofStringToStruct",
+			&struct {
+				Config map[string]basicAppConfig
+			}{},
+			[]*envValue{
+				&envValue{"FOOO", path{"Config", "foo", "StringValue"}},
+				&envValue{"10", path{"Config", "foo", "IntValue"}},
+			},
+			&struct {
+				Config map[string]basicAppConfig
+			}{
+				Config: map[string]basicAppConfig{
+					"foo": basicAppConfig{
+						StringValue: "FOOO",
+						IntValue:    10,
+					},
+				},
+			},
+		},
+		{
+			"WithMapofIntToStruct",
+			&struct {
+				Config map[int]basicAppConfig
+			}{},
+			[]*envValue{
+				&envValue{"FOOO", path{"Config", "0", "StringValue"}},
+				&envValue{"10", path{"Config", "0", "IntValue"}},
+			},
+			&struct {
+				Config map[int]basicAppConfig
+			}{
+				Config: map[int]basicAppConfig{
+					0: basicAppConfig{
+						StringValue: "FOOO",
+						IntValue:    10,
+					},
+				},
+			},
+		},
+		{
+			"WithArrayofInts",
+			&struct {
+				Config []int
+			}{},
+			[]*envValue{
+				&envValue{"1", path{"Config", "0"}},
+				&envValue{"10", path{"Config", "1"}},
+			},
+			&struct {
+				Config []int
+			}{
+				Config: []int{1, 10},
+			},
+		},
+		{
+			"WithArrayofStructs",
+			&struct {
+				Config []basicAppConfig
+			}{},
+			[]*envValue{
+				&envValue{"Test", path{"Config", "0", "StringValue"}},
+				&envValue{"10", path{"Config", "0", "IntValue"}},
+				&envValue{"true", path{"Config", "0", "BoolValue"}},
+				&envValue{"Test2", path{"Config", "1", "StringValue"}},
+				&envValue{"20", path{"Config", "1", "IntValue"}},
+				&envValue{"false", path{"Config", "1", "BoolValue"}},
+			},
+			&struct {
+				Config []basicAppConfig
+			}{
+				Config: []basicAppConfig{
+					{
+						BoolValue:   true,
+						IntValue:    10,
+						StringValue: "Test",
+					},
+					{
+						BoolValue:   false,
+						IntValue:    20,
+						StringValue: "Test2",
+					},
+				},
+			},
+		},
+		{
+			"WithArrayofPointerToStructs",
+			&struct {
+				Config []*basicAppConfig
+			}{},
+			[]*envValue{
+				&envValue{"Test", path{"Config", "0", "StringValue"}},
+				&envValue{"10", path{"Config", "0", "IntValue"}},
+				&envValue{"true", path{"Config", "0", "BoolValue"}},
+				&envValue{"Test2", path{"Config", "1", "StringValue"}},
+				&envValue{"20", path{"Config", "1", "IntValue"}},
+				&envValue{"false", path{"Config", "1", "BoolValue"}},
+			},
+			&struct {
+				Config []*basicAppConfig
+			}{
+				Config: []*basicAppConfig{
+					{
+						BoolValue:   true,
+						IntValue:    10,
+						StringValue: "Test",
+					},
+					{
+						BoolValue:   false,
+						IntValue:    20,
+						StringValue: "Test2",
+					},
+				},
+			},
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.Label, func(t *testing.T) {
+			err := subject.assignValues(reflect.ValueOf(testCase.Value).Elem(), testCase.Values, []string{})
+			if err != nil {
+				t.Logf("Expected no error, got %s", err.Error())
+				t.Fail()
+			}
+
+			assert.Exactly(t, testCase.Expectation, testCase.Value)
```

Comment:
Don't mix stblib test and testify.

Either you use stblib everywhere or testify, but not the both.